Mathlib/Algebra/Order/BigOperators/GroupWithZero/List.lean

```diff
@@ -58,6 +58,21 @@ theorem prod_map_le_prod_map₀ {ι : Type*} {s : List ι} (f : ι → R) (g : 
       simp [ha]
     · apply (h0 _ _).trans (h _ _) <;> simp
 
+theorem prod_map_le_pow_lenght {F L : Type*} [MulPosMono R] [FunLike F L R] [NonnegHomClass F L R]
```

Comment:
I'm guessing you should jut assume `0 <= f y`.
This is more general and you might also be able to avoid the import increase this way.
And probably `f y` could be replace with any arbitrary `y : R`.

Reply:
I would need to assume `0 <= f`, right? I don't think assuming `0 <= f y` is enough.

Reply:
I could write `∀ x : L, 0 ≤ f x` explicitly to avoid using `NonnegHomClass` and changing the imports, but I am not sure whether that would be desirable.

Reply:
Yeah you are right. Probably `0 <= f x <= y` for all `x` in the list should imply `(l.map f).prod <= y ^ l.length`. And this is what I should have said.

Reply:
The reason I somewhat prefer avoiding `NonnegHomClass` here is because i feel like the function here will usually be an unbundled lambda expression and I think it is quite rare that `NonnegHom` actually appears in the application.

Reply:
I have just pushed these changes.